pkg/controller/shared/legacy_kube_informers.go

```diff
@@ -65,6 +65,159 @@ func (f *podInformer) Lister() *cache.StoreToPodLister {
 	return &cache.StoreToPodLister{Indexer: informer.GetIndexer()}
 }
 
+type NodeInformer interface {
+	Informer() framework.SharedIndexInformer
+	Indexer() cache.Indexer
+	Lister() *cache.StoreToNodeLister
+}
+
+type nodeInformer struct {
+	*sharedInformerFactory
+}
+
+func (f *nodeInformer) Informer() framework.SharedIndexInformer {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	informerObj := &kapi.Pod{}
+	informerType := reflect.TypeOf(informerObj)
+	informer, exists := f.informers[informerType]
+	if exists {
+		return informer
+	}
+
+	lw := f.customListerWatchers.GetListerWatcher(kapi.Resource("nodes"))
+	if lw == nil {
+		lw = &cache.ListWatch{
+			ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
+				return f.kubeClient.Nodes().List(options)
+			},
+			WatchFunc: func(options kapi.ListOptions) (watch.Interface, error) {
+				return f.kubeClient.Nodes().Watch(options)
+			},
+		}
+
+	}
+
+	informer = framework.NewSharedIndexInformer(
+		lw,
+		informerObj,
+		f.defaultResync,
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	f.informers[informerType] = informer
+
+	return informer
+}
+
+func (f *nodeInformer) Indexer() cache.Indexer {
+	informer := f.Informer()
+	return informer.GetIndexer()
+}
+
+func (f *nodeInformer) Lister() *cache.StoreToNodeLister {
+	informer := f.Informer()
+	return &cache.StoreToNodeLister{Store: informer.GetStore()}
+}
+
+type PersistentVolumeInformer interface {
+	Informer() framework.SharedIndexInformer
+	Indexer() cache.Indexer
+}
```

Comment:
No one is listing them yet.

On Thu, Jul 7, 2016 at 9:45 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/controller/shared/legacy_kube_informers.go
> <https://github.com/openshift/origin/pull/9578#discussion_r69908418>:
>
> > +}
> > +
> > +func (f *nodeInformer) Indexer() cache.Indexer {
> > +	informer := f.Informer()
> > +	return informer.GetIndexer()
> > +}
> > +
> > +func (f *nodeInformer) Lister() *cache.StoreToNodeLister {
> > +	informer := f.Informer()
> > +	return &cache.StoreToNodeLister{Store: informer.GetStore()}
> > +}
> > +
> > +type PersistentVolumeInformer interface {
> > +	Informer() framework.SharedIndexInformer
> > +	Indexer() cache.Indexer
> > +}
>
> Why this and the PVCInformer don't need a Lister()?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9578/files/849a565571da716634dcbbb80b64363885fb8448..7d64c52d5c42cc6aacf62c585d0032015fc6e853#r69908418>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_py2UQWKNDsXz0iJNtiChF22n9Milks5qTQMKgaJpZM4I-xTE>
> .
>